lib/render/RayCaster.cpp

```diff
@@ -264,16 +303,16 @@ bool RayCaster::UserCoordinates::IsUpToDate( const RayCasterParams* params,
             ( dims[i]   != gridDims[i]      )   )
         {
             delete grid;
-            return false;
+            return 1;
         }
     }
 
     // now we know it's up to date!
     delete grid;
-    return true;
+    return 0;
 }
         
-bool RayCaster::UserCoordinates::UpdateCoordinates( const RayCasterParams* params,
+int  RayCaster::UserCoordinates::UpdateFaceAndData( const RayCasterParams* params,
```

Comment:
This is an awfully long function with a lot of repetitive operations. How about creating a helper method parameterized by the face to perform the loading of the face data?

Reply:
This function is long but it's also got clear structures, so a reader can get understand it very easily. 
Plus, within this function, though each block seems repetitive, they aren't, because the face indices are different and they have to be hard coded before we utilize some real fancy parameterization. 
What do you think?

Reply:
You could replace the 6 nested loop pairs with 3 calls to functions that looked like the snippet below. This would reduce the repetitive code, and make UpdateFaceAndData shorter and easier to understand.

`float *updateLeftRight(
    const vector <size_t> &dims,
    bool zMinFlag
) const {

    float buf[3];
    size_t z = zMinFlag ? 0 : dims[2] - 1;

    float *face = new float[ dims[0] * dims[1] * 3 ];
    size_t idx = 0;
    for( size_t y = 0; y < dims[1]; y++ ) {
        for( size_t x = 0; x < dims[0]; x++ )
        {
            grid->GetUserCoordinates( x, y, z, buf[0], buf[1], buf[2] );
            frontFace[ idx++  ] = (float)buf[0];
            frontFace[ idx++  ] = (float)buf[1];
            frontFace[ idx++  ] = (float)buf[2];
        }
    }
    return(face);
}

float *updateFrontBack(
    const vector <size_t> &dims,
    bool zMinFlag
) const {
...
}

float *updateBottomTop(
    const vector <size_t> &dims,
    bool zMinFlag
) const {
...
}`

Reply:
I have to disagree. I found this function very confusing and difficult to understand.

Reply:
Make it from 6 loops to 3 functions

Reply:
addressed in 283f503a